### ISSUE TITLE
best solution for 3D / 2D models data settings

### ISSUE THREAD

**Issue by thegodone** [aimat-lab/gcnn_keras#57]
for preloaded data we don't have this issue but for new data that would be nice to have a clean way to know what type of data process we need to put in place based on the architecture.

This also include some model / architecture specification like radius or normalization etc... so all meta data transformation applied to make dynamically "architecture + data preprocess" for a given new dataset.

=> I would suggest to store a "3D" boolean parameter:  true / false in data json part for each model.

**Comment by PatReis** [comment]
We can put the `prepare_data` and `read_in_memory` together with `set_attributes` in the hyper-parameter in the methods list. Then each model can decide whether to generate structures or not. I can modify the last `keras-gcn.py` you sent me, if that works with yout environment. The conformer generation runs parallel and should be efficient for up to ~1-10 Mio. molecules if you have both openbabel and rdkit installed.

**Comment by thegodone** [comment]
Solved by another issue
